kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/RestSpecManagementServiceImpl.java

```diff
@@ -90,7 +90,7 @@ public Response saveContainerSpec(@Context HttpHeaders headers,
             @ApiResponse(code = 400, message = "Controller exception"),
             @ApiResponse(code = 500, message = "Unexpected error")
     })
-    @POST
+    @PUT
```

Comment:
That breaks backward compatibility.
What is the reason for such change?

Reply:
Hi @sutaakar If you see the code inside the POST & PUT methods currently we have opposite implementations. This change fixes the readability and API definition.
^ @cristianonicolai

Reply:
So, you want to use POST to create a resource identified explicitly bu URL?
The backward compatibility issue I meant that if users use the POST endpoint to update containers in server template then this change will break it, forcing user to update the implementation.

Reply:
@sutaakar I agree this would need users to change the implementation in the existing applications, but I think it is also important to fix the API definition with the correct methods. 
Hi @cristianonicolai, Can you please share your thoughts on this?
Thanks

Reply:
@akumar074 I think it is a valid concern, although I agree we should probably change the annotations, this is very likely to break users of the API. If we look in the original JIRA, this change wouldn't affect the reported issue in any way, so doing this change would be only recommended if extremely necessary. My 2 cents